tests/integration/test_pluggable_integration.py

```diff
@@ -0,0 +1,247 @@
+                    assert original_condition.label == fetched_condition.label
+                    assert original_condition.condition_type == fetched_condition.condition_type
+                    assert len(original_condition.matchers) == len(fetched_condition.matchers)
+                    assert len(original_condition.partitions) == len(fetched_condition.partitions)
+        finally:
+            [adapter.delete(key) for key in ['SPLITIO.split.sample_feature',
+                'SPLITIO.splits.till',
+                'SPLITIO.split.all_feature',
+                'SPLITIO.split.killed_feature',
+                'SPLITIO.split.Risk_Max_Deductible',
+                'SPLITIO.split.whitelist_feature',
+                'SPLITIO.split.regex_test',
+                'SPLITIO.split.boolean_test',
+                'SPLITIO.split.dependency_test']]
+
+
+class PluggableSegmentStorageIntegrationTests(object):
+    """Pluggable Segment storage e2e tests."""
+
+    def test_put_fetch_contains(self):
+        """Test storing and retrieving splits in pluggable."""
+        adapter = StorageMockAdapter()
+        try:
+            storage = PluggableSegmentStorage(adapter)
+            adapter.set(storage._prefix.format(segment_name='some_segment'), {'key1', 'key2', 'key3', 'key4'})
+            adapter.set(storage._segment_till_prefix.format(segment_name='some_segment'), 123)
+            assert storage.segment_contains('some_segment', 'key0') is False
+            assert storage.segment_contains('some_segment', 'key1') is True
+            assert storage.segment_contains('some_segment', 'key2') is True
+            assert storage.segment_contains('some_segment', 'key3') is True
+            assert storage.segment_contains('some_segment', 'key4') is True
+            assert storage.segment_contains('some_segment', 'key5') is False
+
+            fetched = storage.get('some_segment')
+            assert fetched.keys == set(['key1', 'key2', 'key3', 'key4'])
+            assert fetched.change_number == 123
+        finally:
+            adapter.delete('SPLITIO.segment.some_segment')
+            adapter.delete('SPLITIO.segment.some_segment.till')
+
+
+class PluggableImpressionsStorageIntegrationTests(object):
+    """Pluggable Impressions storage e2e tests."""
+
+    def _put_impressions(self, adapter, metadata):
+        storage = PluggableImpressionsStorage(adapter, metadata)
+        storage.put([
+            impressions.Impression('key1', 'feature1', 'on', 'l1', 123456, 'b1', 321654),
+            impressions.Impression('key2', 'feature1', 'on', 'l1', 123456, 'b1', 321654),
+            impressions.Impression('key3', 'feature1', 'on', 'l1', 123456, 'b1', 321654)
+        ])
+
+
+    def test_put_fetch_contains(self):
+        """Test storing and retrieving splits in pluggable."""
+        adapter = StorageMockAdapter()
+        try:
+            self._put_impressions(adapter, get_metadata({}))
+
+            imps = adapter.get('SPLITIO.impressions')
```

Comment:
it should be `imps = adapter.pop_items('SPLITIO.impressions')`

For completeness, I would add `pop_items` to your wrapper interface, although you will not use it at the moment, because it is used by the producer method (`pop_many`) of impressions and events pluggable storages.

`pop_items` retrieves the items added via `push_items` operation. Together with `get_items_count`, these 3 operations are "list" operation types, equivalent to  `rpop`, `rpush` and `llen` in Redis respectively.

Reply:
sure, will do